core/src/main/scala/kafka/controller/KafkaController.scala

```diff
@@ -1162,9 +1162,14 @@ class KafkaController(val config: KafkaConfig, zkUtils: ZkUtils, val brokerState
     @throws[Exception]
     def handleNewSession() {
       info("ZK expired; shut down all controller components and try to re-elect")
-      onControllerResignation()
-      inLock(controllerContext.controllerLock) {
-        controllerElector.elect
+      if (controllerElector.getControllerID() != config.brokerId) {
+        onControllerResignation()
+        inLock(controllerContext.controllerLock) {
+          controllerElector.elect
+        }
+      } else {
+        //maybe create by current session or the previous zk session's ephemeral node is not deleted
```

Comment:
@pengwei-li : The zookeeper people have confirmed that a client shouldn't see ephmeral nodes created by the expired session in a newly established session.  http://mail-archives.apache.org/mod_mbox/zookeeper-user/201701.mbox/%3CB512F6DE-C0BF-45CE-8102-6F242988268E%40apache.org%3E